src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCheckBoxCell.DataGridViewCheckBoxCellRenderer.cs

```diff
@@ -13,23 +13,35 @@ private static class DataGridViewCheckBoxCellRenderer
         {
             private static VisualStyleRenderer? s_visualStyleRenderer;
 
-            public static VisualStyleRenderer CheckBoxRenderer
+            public static VisualStyleRenderer CheckBoxRenderer => s_visualStyleRenderer ??= new VisualStyleRenderer(CheckBoxElement);
+
+            public static void DrawCheckBox(Graphics g, Rectangle bounds, int state)
             {
-                get
-                {
-                    if (s_visualStyleRenderer is null)
-                    {
-                        s_visualStyleRenderer = new VisualStyleRenderer(CheckBoxElement);
-                    }
+                CheckBoxRenderer.SetParameters(CheckBoxElement.ClassName, CheckBoxElement.Part, state);
+                CheckBoxRenderer.DrawBackground(g, bounds, Rectangle.Truncate(g.ClipBounds));
 
-                    return s_visualStyleRenderer;
+                // It uses because the checked style of the checkbox in a selected cell has a low contrast ratio with the background. Only for Windows 11.
+                // State 5 is selected, unfocused.
+                // State 6 is selected, focused.
+                if (OsVersion.IsWindows11_OrGreater && (state == 5 || state == 6))
```

Comment:
What do you think about this?
```suggestion
                // It uses because the checked style of the checkbox in a selected cell has a low contrast ratio with the background. Only for Windows 11.
                if (OsVersion.IsWindows11_OrGreater && (state == /* selected, unfocused */5 || state == /* selected, focused */6))
```
...or even better:
```suggestion
                // It uses because the checked style of the checkbox in a selected cell has a low contrast ratio with the background. Only for Windows 11.
                const int StateSelectedUnfocused = 5;
                const int StateSelectedFocused = 6;
                if (OsVersion.IsWindows11_OrGreater && (state == StateSelectedUnfocused  || state ==StateSelectedFocused))
```

Reply:
Looks like I didn't get the spacing right, as I've written it in the browser.